### DISCUSSION TITLE
How to scale kiali

### DISCUSSION THREAD

**Discussion by chris530** [kiali/kiali#4272] [General]
Kiali works great for us on our smaller clusters,  but  generating graphs fails on our larger namespaces with 1k nodes.    Any suggestions on scaling kiali ?   I have tried the obvious things like increase resources and replicas.

**Comment by jmazzitelli** [comment]
Good question that is actively under discussion - https://github.com/kiali/kiali/discussions/4080

**Comment by israel-hdez** [comment]
I wonder if you could tell:
* How many namespaces do you have?
* How many workloads do you have?
* Are you trying to get a graph of the whole cluster, or a graph of a (some) specific namespace(s)?
* What failure are you seeing in Kiali? (perhaps provide the error message in the UI or errors in logs?)

**Comment by lucasponce** [comment]
> Kiali works great for us on our smaller clusters, but generating graphs fails on our larger namespaces with 1k nodes. Any suggestions on scaling kiali ? I have tried the obvious things like increase resources and replicas.

Thanks @chris530, also worth to mention which version are you using.
There are some performance improvements that may vary depending of it.

But yes, I guess your comment is directly related to the https://github.com/kiali/kiali/discussions/4080 discussion

**Comment by lucasponce** [comment]
I've converted to a discussion as it won't have a direct action on it until the discussion of #4080 moves.

I have one question for @chris530, one of the limitations of Kiali, or in general any admin tool that deals with Kubernetes objects in large cluster is the cluster permissions.

If Kiali could be deployed with cluster permissions, we could use that design premise to reduce all namespaces queries into one, and that would have a significant improvement in the performance.

Today, it's not available, but if that option could be valid for some users under special situations like this, that may help in the way that all kubernetes objects are fetched, today they overload in general the API and that may impact into general performance.

Do you have other admins tools that can be granted with cluster permissions @chris530 and that may be accepted into some scenarios ?

**Comment by hhovsepy** [comment]
@chris530 if you are still using Kiali, significant performance improvements have been made starting with Kiali v1.85. For more details, visit: [Kiali Performance FAQ](https://kiali.io/docs/faq/performance/)
